site2/docs/schema-evolution-compatibility.md

```diff
@@ -0,0 +1,165 @@
+
+### How Pulsar schema should evolve? 
+
+The answer is Pulsar schema compatibility check strategy. It determines how schema compares old schemas with new schemas in topics.
+
+For more information, see [Schema compatibility check strategy](#schema-compatibility-check-strategy).
+
+### How does Pulsar support schema evolution?
+
+1. When a producer/consumer/reader connects to a broker, the broker deploys the schema compatibility checker configured by `schemaRegistryCompatibilityCheckers` to enforce schema compatibility check. 
+
+    The schema compatibility checker is one instance per schema type. 
+    
+    Currently, Avro and JSON have their own compatibility checkers, while all the other schema types share the default compatibility checker which disables schema evolution.
+
+2. The producer/consumer/reader sends its client `SchemaInfo` to the broker. 
+   
+3. The broker knows the schema type and locates the schema compatibility checker for that type. 
+
+4. The broker uses the checker to check if the `SchemaInfo` is compatible with the latest schema of the topic by applying its compatibility check strategy. 
+   
+   Currently, the compatibility check strategy is configured at the namespace level and applied to all the topics within that namespace.
+
+## Schema compatibility check strategy
+
+Pulsar has 8 schema compatibility check strategies, which are summarized in the following table.
+
+Suppose that you have a topic containing three schemas (V1, V2, and V3), V1 is the oldest and V3 is the latest:
+
+| Compatibility check strategy | Definition | Changes allowed | Check against which schema | Upgrade first |
+|---|---|---|---|---|
+| `ALWAYS_COMPATIBLE` | Disable schema compatibility check | All changes are allowed | All previous versions | Any order |
+| `ALWAYS_INCOMPATIBLE` | Disable schema evolution | All changes are disabled | / | / |
+| `BACKWARD` | Consumers using the schema V3 can process data written by producers using the schema V3 or V2 | Add optional fields<br>Delete fields | Latest version | Consumers |
+| `BACKWARD_TRANSITIVE` | Consumers using the schema V3 can process data written by producers using the schema V3, V2 or V1 | Add optional fields<br>Delete fields | All previous versions | Consumers |
+| `FORWARD` | Consumers using the schema V3 or V2 can process data written by producers using the schema V3 | Add fields<br>Delete optional fields | Latest version | Producers |
+| `FORWARD_TRANSITIVE` | Consumers using the schema V3, V2 or V1 can process data written by producers using the schema V3 | Add fields<br>Delete optional fields | All previous versions | Producers |
+| `FULL` | Backward and forward compatible between the schema V3 and V2 | Modify optional fields | Latest version | Any order |
+| `FULL_TRANSITIVE` | Backward and forward compatible among the schema V3, V2, and V1 | Modify optional fields | All previous versions | Any order |
+
```

Comment:
@sijie For this table, I add the **Definition** column, are the contents accurate?

Reply:
looks good to me.